src/Atmos/Model/AtmosModel.jl

```diff
@@ -716,4 +718,107 @@ function init_state_conservative!(
 )
     m.init_state_conservative(m, state, aux, coords, t, args...)
 end
+
+struct RoeNumericalFlux <: NumericalFluxFirstOrder end
+function numerical_flux_first_order!(
+    numerical_flux::RoeNumericalFlux,
+    balance_law::AtmosModel,
+    fluxᵀn::Vars{S},
+    normal_vector::SVector,
+    state_conservative⁻::Vars{S},
+    state_auxiliary⁻::Vars{A},
+    state_conservative⁺::Vars{S},
+    state_auxiliary⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        balance_law,
+        fluxᵀn,
+        normal_vector,
+        state_conservative⁻,
+        state_auxiliary⁻,
+        state_conservative⁺,
+        state_auxiliary⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+    param_set = balance_law.param_set
+    _cv_d::FT = cv_d(param_set)
+    _T_0::FT = T_0(param_set)
+    γ::FT = cp_d(param_set) / cv_d(param_set)
+
+    Φ = gravitational_potential(balance_law, state_auxiliary⁻)
+
+    ρ⁻ = state_conservative⁻.ρ
+    ρu⁻ = state_conservative⁻.ρu
+    ρe⁻ = state_conservative⁻.ρe
+
+    u⁻ = ρu⁻ / ρ⁻
+    p⁻ = pressure(
+        balance_law,
+        balance_law.moisture,
+        state_conservative⁻,
+        state_auxiliary⁻,
+    )
+    h⁻ = (ρe⁻ + p⁻) / ρ⁻
+
+    ρ⁺ = state_conservative⁺.ρ
+    ρu⁺ = state_conservative⁺.ρu
+    ρe⁺ = state_conservative⁺.ρe
+
+    u⁺ = ρu⁺ / ρ⁺
+    p⁺ = pressure(
+        balance_law,
+        balance_law.moisture,
+        state_conservative⁺,
+        state_auxiliary⁺,
+    )
+    h⁺ = (ρe⁺ + p⁺) / ρ⁺
+
+    ũ = (sqrt(ρ⁻) * u⁻ + sqrt(ρ⁺) * u⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+    h̃ = (sqrt(ρ⁻) * h⁻ + sqrt(ρ⁺) * h⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+    c̃ = sqrt((γ - 1) * (h̃ - ũ' * ũ / 2 - Φ + _cv_d * _T_0))
+
+    # chosen by fair dice roll
+    # guaranteed to be random
+    ω = FT(π) / 3
+    δ = FT(π) / 5
+    random_unit_vector = SVector(sin(ω) * cos(δ), cos(ω) * cos(δ), sin(δ))
+
+    # tangent space basis
+    τ1 = random_unit_vector × normal_vector
+    τ2 = τ1 × normal_vector
+
+    ũᵀn = ũ' * normal_vector
+    ũc̃⁻ = ũ + c̃ * normal_vector
+    ũc̃⁺ = ũ - c̃ * normal_vector
+
+    Λ = SDiagonal(
+        abs(ũᵀn - c̃),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn + c̃),
+    )
+
+    M = hcat(
+        SVector(1, ũc̃⁻[1], ũc̃⁻[2], ũc̃⁻[3], h̃ - c̃ * ũᵀn),
+        SVector(0, τ1[1], τ1[2], τ1[3], τ1' * ũ),
+        SVector(0, τ2[1], τ2[2], τ2[3], τ2' * ũ),
+        SVector(1, ũ[1], ũ[2], ũ[3], ũ' * ũ / 2 + Φ - _T_0 * _cv_d),
```

Comment:
This looks odd. It seems it depends on the arbitrary absolute temperature scale `_T_0`, which it should not depend on (this is a reference scale that should drop out of all dynamics).

Reply:
Shouldn't the last term here be only `ũ' * ũ / 2`? The quantity that is there, adding potential energy and subtracting an arbitrary reference internal energy, is not a meaningful thermodynamic quantity. For dry dynamics, everything should be completely independent of `T_0`, which is arbitrary (the dynamics are unchanged for `T_0 = 0` or `T_0 = 1e12`). So it's difficult to see how `T_0` arises here. (While the linearized pressure depends on `T_0`, only pressure gradients matter dynamically, and those do not depend on `T_0`.)

@mwarusz: Could you please provide a reference from which you are working here?